pkg/collectsub/datasource/filesource/filesource.go

```diff
@@ -115,6 +116,12 @@ func (d *fileDataSources) DataSourcesUpdate() (<-chan error, error) {
 				updateChan <- err
 				fmt.Printf("got err while watching: %+v\n", err)
 				return
+
+			case <-ctx.Done():
+				err := fmt.Errorf("file watcher ending from context closure")
+				updateChan <- err
+				fmt.Printf("ctx is closed: %+v\n", err)
```

Comment:
Recommend to use log instead of fmt.

Reply:
+1 , will remove these statements since the errors are propagated and would be handled elsewhere

Reply:
👍